src/disk.hpp

```diff
@@ -42,45 +41,42 @@ class Disk {
 
     virtual std::string GetFileName() = 0;
 
-    virtual ~Disk(){};
+    virtual ~Disk() = default;
 };
 
-class FileDisk : public Disk {
-public:
-    inline explicit FileDisk(const fs::path &filename)
+struct FileDisk : Disk {
+    explicit FileDisk(const fs::path &filename)
     {
         filename_ = filename;
 
         // Opens the file for reading and writing
-        f_ = fopen(filename.c_str(), "w+b");
+        f_ = ::fopen(filename.c_str(), "w+b");
+        if (f_ == nullptr) {
+            throw InvalidValueException(
+                "Could not open " + filename.string() + ": " + ::strerror(errno));
+        }
     }
 
     FileDisk(FileDisk &&fd)
     {
         filename_ = fd.filename_;
         f_ = fd.f_;
-        fd.f_ = NULL;
+        fd.f_ = nullptr;
     }
 
-    bool isOpen() { return (f_ != NULL); }
```

Comment:
is open was only used in one place, to check if the file opened successfully, and throw if it didn't. So I moved that directly into the constructor instead.